### PR TITLE
Spec for gluster cluster list view

### PR DESCRIPTION
**Pull request by gnehapk**
tendrl-bug-id: #181

### CODE REVIEW

File: specs/gluster_cluster_list_view.adoc [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+=== Data model impact:
+See 'Impacted Modules' section where all the details are populated.
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+/api/1.0/GetClusterList - fetch the list of clusters
+
+Sample API Response - required.
```
Comment (gnehapk) [inline specs/gluster_cluster_list_view.adoc:85]: @anivargi @brainfunked @nthomas-redhat As the cluster list will list all the clusters which are already imported and ready to import, I think it make sense to have a single API call for it. As per the current API, cluster which are available for importing is present inside GetNodeList API and imported cluster is in GetClusterList API.

**Comment by gnehapk** [comment]
@brainfunked @nthomas-redhat @shtripat @anivargi @cloudbehl @a2batic Please review.

File: specs/gluster_cluster_list_view.adoc (hunk 1 above)
Comment (anupnivargi) [inline specs/gluster_cluster_list_view.adoc]: @gnehapk create a issue in API for this change

Comment (anivargi) [reply specs/gluster_cluster_list_view.adoc:85]: @gnehapk add a issue on Tendrl/api

File: specs/gluster_cluster_list_view.adoc [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+* Import button(if ready to import)
+* Launch Dashboard button(if already imported)
+* Cluster Details button(if already imported)
+* Message to state cluster configurable state for eg. ready to import, cluster misconfigured.
+
+The workflow is as follows:
+
+* API will provide the list of clusters to the Tendrl.
+
+* User can click on expand icon of each cluster record to see the error messages/logs and hosts associated with it.
```
Comment (gnehapk) [inline specs/gluster_cluster_list_view.adoc]: @brainfunked @anivargi @nthomas-redhat @r0h4n What messages will be displayed in the "Message" tab if cluster is imported and ready to use?

File: specs/gluster_cluster_list_view.adoc [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+* Status
+* Cluster Name
+* Cluster Version
+
+Required from API - 
+
+* Managed State
+* Message to state cluster configurable for eg. ready to import, cluster misconfigured.
+
```
Comment (gnehapk) [inline specs/gluster_cluster_list_view.adoc]: @anivargi @brainfunked @nthomas-redhat @r0h4n  Host IP address is also required for displaying it under "Hosts" tab of expanded view. Currently in the API its not coming for all node.

File: specs/gluster_cluster_list_view.adoc (hunk 2 above)
Comment (cybernth) [reply specs/gluster_cluster_list_view.adoc]: We can display events/notifications associated with that cluster

File: specs/gluster_cluster_list_view.adoc [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+* In the hosts tab, host name and IP address will be displayed.
+
+* Based on the managed state, import/launch dasboard/cluster details buttons will be displayed for each cluster.
+  * If Managed state is "Yes", Launch dashboard and cluster details button will be displayed.
+  * If Managed state is "No" and cluster is ready to import, Import button will be displayed.
+  * If Managed state is "No" and cluster is not properly configured, Imported will be disabled.
+
+* Clicking Import button will take to import cluster workflow.
+
+* Clicking Launch dashboard button will launch the dashboard in a new window.
```
Comment (cybernth) [inline specs/gluster_cluster_list_view.adoc]: please clarify what you mean by new window. popup? or new tab in the browser? Also link with grafana dashboard spec

File: specs/gluster_cluster_list_view.adoc [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+* Clicking Cluster details button will take to cluster objects details page.
```
Comment (cybernth) [inline specs/gluster_cluster_list_view.adoc]: what is the content of cluster object details? do you have a wireframe?

File: specs/gluster_cluster_list_view.adoc [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+* Create a view for listing clusters.
+* Use the patternfly expanded list views design to display list of clusters.
+* Implement client side filer and sorting functionality using angular's filter and orderby components.
+* Send polling call after 10 seconds to get the updated data. 
+* Show total count of clusters present in Tendrl.
+
+Tendrl API related changes:
+
+* Get API call for getting cluster list.
```
Comment (cybernth) [inline specs/gluster_cluster_list_view.adoc:68]: Please mention the get cluster list API changes required

File: specs/gluster_cluster_list_view.adoc (hunk 3 above)
Comment (cybernth) [reply specs/gluster_cluster_list_view.adoc]: As I mentioned earlier, please provide the details about `detected cluster` object and corresponding changes to APIs. either you can do it here or create a separate issue to track if that doesn't exist and link to this spec. I know this is not commited from backend for firest drop. if that is the case, what you are planing to implement in the first drop?

File: specs/gluster_cluster_list_view.adoc (hunk 4 above)
Comment (gnehapk) [reply specs/gluster_cluster_list_view.adoc]: @nthomas-redhat Its a new tab in browser.

File: specs/gluster_cluster_list_view.adoc (hunk 5 above)
Comment (gnehapk) [reply specs/gluster_cluster_list_view.adoc]: @nthomas-redhat Content of cluster object details will be handled in a separate spec as its a separate view.

**Review by cybernth — APPROVED** [review]
LGTM

**Comment by gnehapk** [comment]
For the drop targeted on 4th Aug, following items will be supported from the spec -
- Cluster list will contain only the imported cluster
- Only hosts tab will be present inside expanded cluster view
- Host Name will be displayed for each host.

**Review by brainfunked — APPROVED** [review]

**Pull request merged**